src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -1467,24 +1470,24 @@ App::DocumentObjectExecReturn *Hole::execute(void)
             const double pi = 3.14159265359;
             bool leftHanded = (bool) ThreadDirection.getValue();
 
-            // Nomenclature and formulae according to Schaeffler Technical pocket guide (edition 2018)
-            // this is the same for all metric and UTS threads.
-            double D = threadDescription[ThreadType.getValue()][ThreadSize.getValue()].diameter;
-            double P = ThreadPitch.getValue();
-            double H1 = 0.54127 * P;
-            double H = 0.86603 * P;
-            double D1 = D - 2 * H1;
-            double alfa = 60.0 * pi/180;
-
-
-            double clearance;   // clearance used for Actual Thread modeling.
+            // Nomenclature and formulae according to Figure 1 of ISO 68-1
+            // this is the same for all metric and UTS threads as stated here:
+            // https://en.wikipedia.org/wiki/File:ISO_and_UTS_Thread_Dimensions.svg
+            // Note that in the ISO standard, Dmaj is called D, which has been followed here.
+            double D = threadDescription[ThreadType.getValue()][ThreadSize.getValue()].diameter;  //
+            double P = ThreadPitch.getValue();            // Pitch
+            double H = sqrt(3.0)/2.0 * P;                 // Height of fundamentral triangle
+            double H1 = 5*H/8 * P;                        // The thread depth
+            double D1 = D - 2 * H1;                       // Basic minor diameter of internal thread
+            double alfa = Base::toRadians(60.0);          // Angle of the fundamental triangle. Same for both ISO and UTS.
+
+            double clearance;                             // clearance to be added on the diameter
             if (UseCustomThreadClearance.getValue())
                 clearance = CustomThreadClearance.getValue();
             else
                 clearance = getThreadClassClearance();
 
-
-            // construct the cross section
+            // construct the cross section going counter-clockwise
             gp_Pnt p1 = toPnt(xDir * D1/2.0 + P*zDir + clearance/2.0*xDir);
             gp_Pnt p2 = toPnt(toVec(p1) + (D-D1)/2.0*xDir + (D-D1)/2.0*tan(alfa/2.0)*zDir);

```

Comment:
Despite we have no hard code style rule, we commonly use spaces around operators. So in this case:

```
gp_Pnt p2 = toPnt(toVec(p1) + (D - D1) / 2.0 * xDir
                  + (D - D1) / 2.0 * tan(alfa / 2.0) * zDir);
```

Reply:
Yikes :-o
Maybe we should have a style guide

Reply:
> Maybe we should have a style guide

Yes, however, the consensus in the recent discussions in the forum is to use the style of the surrounding code. Just please do me the favor and add spaced also around the * operator - its simply better readable.

Reply:
Fine, I can live with that.

Reply:
done